### ISSUE TITLE
Closed-loop evaluation result on Bench2Drive

### ISSUE THREAD

**Issue by questionstorer** [RenzKa/simlingo#43]
Thank you for this execellent work!
We managed to run the closed-loop evaluation result on Bench2Drive using your code with some debug process. We can run all the routes except for one route mentioned [here](https://github.com/RenzKa/simlingo/issues/42). Later we compute the overall DS and SR using the merge_route_json.py based on these 219 routes. 

We get a DS of **75.50** and SR of **67.12%**. The DS is significantly lower compared to the result in the paper which is **85.94**. May I ask that is this result expected?

**Comment by RenzKa** [comment]
Cool thanks for the detailed information.
What was the problem when using the Bench2Drive folder?

In the Bench2Drive folder are some custom adjustments for the benchmark, so this might be the reason why you could not reproduce the results.

I tested the released model with two evaluation seeds and got:
Seed 0:  DS: 87.40,  SR: 67.74
Seed 1:  DS: 87.67,  SR: 70.64

**Comment by questionstorer** [comment]
> Sounds great. Let me know if there are issues with the bench2drive folders.

Looks like after the last commit, it can be used. But still we have to fix some lines in the `agent_simlingo.py` as described above.
Now we can reproduce the result, although the route 111 still crashes. We get a DS of **86.53** and SR of **67.12**. Thanks for your comments.

**Comment by jays-nd** [comment]
> Cool thanks for the detailed information. What was the problem when using the Bench2Drive folder?
> 
> In the Bench2Drive folder are some custom adjustments for the benchmark, so this might be the reason why you could not reproduce the results.
> 
> I tested the released model with two evaluation seeds and got: Seed 0: DS: 87.40, SR: 67.74 Seed 1: DS: 87.67, SR: 70.64

Hi Katrin,

I just had a small doubt regarding the released model. Does the training data of these models include Town 13 or the Town 13 is withheld during training?

Thanks,
Jay

**Comment by RenzKa** [comment]
Hi,
Same as the original Bench2Drive dataset and models, we do include Town13 during training. 

Best,
Katrin

**Comment by jays-nd** [comment]
> Hi, Same as the original Bench2Drive dataset and models, we do include Town13 during training.
> 
> Best, Katrin

Thanks! Also, just to confirm, the Bench2Drive closed-loop evaluation scores of "simlingo-base" and "simlingo" in the simlingo paper  (Table 2)(https://arxiv.org/pdf/2503.09594) are based on a model trained on a validation set of Town 13?

Sorry for taking for your time, I should've asked this earlier.

Best,
Jay

**Comment by HuangChiEn** [comment]
Add new reproduction of pretrained checkpoint 👍 
We can successfully run all 220 routes :
simlingo ckpt "epoch=013.ckpt" : 
DS : 88.7, SR : 71.8

<img width="558" height="226" alt="Image" src="https://github.com/user-attachments/assets/2c496a3f-2406-4a42-b250-9fe71b3cd868" />

**Comment by meibohuhu** [comment]
Yes i also ran with results in driving score around 88, success rate around 0.71.
